synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,170 @@
+from typing import TYPE_CHECKING, AsyncGenerator, Dict, List, Optional, Union
+
+from synapseclient.api.json_schema_services import (
+    bind_json_schema_to_entity,
+    delete_json_schema_from_entity,
+    get_invalid_json_schema_validation,
+    get_json_schema_derived_keys,
+    get_json_schema_from_entity,
+    get_json_schema_validation_statistics,
+    validate_entity_with_json_schema,
+)
+from synapseclient.core.async_utils import async_to_sync
+
+if TYPE_CHECKING:
+    from synapseclient import Synapse
+
+
+@async_to_sync
+class JSONSchema:
+    """
+    Mixin class to provide JSON schema functionality.
+    This class is intended to be used with classes that represent Synapse entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the entity.
+    """
+
+    id: Optional[str] = None
+
+    async def bind_json_schema_to_entity_async(
+        self, json_schema_uri: str, *, synapse_client: Optional["Synapse"] = None
+    ) -> Dict[str, Union[str, int, bool]]:
+        """Bind a JSON schema to an entity"""
+        return await bind_json_schema_to_entity(
+            synapse_id=self.id,
+            json_schema_uri=json_schema_uri,
+            synapse_client=synapse_client,
+        )
+
+    async def get_json_schema_from_entity_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> Dict[str, Union[str, int, bool]]:
```

Comment:
For any of the return types of these functions try to convert them into a python dataclass with the `snake_case` format for attributes. That way someone that calls the function would be able to do:

```
(await my_folder.get_json_schema_from_entity_async()).json_schema_version_info.organization_id
```

We want to be very clear for users of the python client what fields are possible to access when they call these methods

Reply:
Thank you for the suggestion! I will make the update.

Reply:
Yes that looks correct (Any the same pattern can be applied to other method calls too). The only thing to note: https://github.com/Sage-Bionetworks/synapsePythonClient/pull/1205#discussion_r2136095677


It's probably not the best example that this API only has a single field: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/annotation/v2/Keys.html - However, by returning a dataclass object we are future proofing this method call to support adding additional fields (If Synapse ever does) to the return method without breaking existing code, since it would be NEW data added to the existing object.